src/GraphQL.MemoryCache/MemoryDocumentCache.cs

```diff
@@ -0,0 +1,71 @@
+using System;
+using GraphQL.Language.AST;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Options;
+
+namespace GraphQL.Caching
+{
+    /// <summary>
+    /// A basic implementation of a document cache, limited by a configured amount of memory.
+    /// </summary>
+    public class MemoryDocumentCache : IDocumentCache, IDisposable
+    {
+        private readonly IMemoryCache _memoryCache;
+        private readonly MemoryDocumentCacheOptions _options;
+
+        /// <summary>
+        /// Initializes a new instance with the default options: 100,000 maximum total query size and no expiration time.
+        /// </summary>
+        public MemoryDocumentCache()
+            : this(new MemoryDocumentCacheOptions())
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance with the specified options.
+        /// </summary>
+        /// <param name="options">A value containing the <see cref="MemoryDocumentCacheOptions"/> to use.</param>
+        public MemoryDocumentCache(IOptions<MemoryDocumentCacheOptions> options)
+            : this(
+                new MemoryCache(new MemoryCacheOptions { SizeLimit = options.Value.MaxTotalQueryLength }),
```

Comment:
```suggestion
                new MemoryCache(options),
```

Reply:
These are not the same class.

Reply:
https://github.com/graphql-dotnet/graphql-dotnet/pull/2173#discussion_r557690734
![изображение](https://user-images.githubusercontent.com/21261007/104656308-24789280-56d0-11eb-8ef8-c0ecb4c12c7f.png)

I didn't even notice 😄

Reply:
We can..... but members are not virtual so we cannot override `SizeLimit` with a meaningful xml comment.

Personally I don't like inheriting option classes.  It caused problems in the serialization libraries here for the same reason.

Do you want to add a `MemoryCacheOptions` property?

Reply:
I really don't care much but it's just a bad design in my opinion.

Reply:
>  so we cannot override SizeLimit with a meaningful xml comment.

No problem.

>  It caused problems in the serialization libraries here for the same reason.

What problems? I suggested inheriting because only one new property actually needed - SlidingExpiration. This property already exists on the `CacheEntry` level but we "pick it up" to ctor options level so we can access it from `GetMemoryCacheEntryOptions`.

Reply:
> No problem.

?

Reply:
I mean that the inability to write a comment should not be an obstacle to avoiding duplicate meaning of options. These are not new options, but an extension of existing ones. We could use `MemoryCacheOptions` directly if not `SlidingExpiration` setting.